backend/editor/controllers/search_controller.py

```diff
@@ -0,0 +1,374 @@
+import math
+from abc import ABC, abstractmethod
+from dataclasses import dataclass
+from typing import Literal
+
+from openfoodfacts_taxonomy_parser import utils as parser_utils
+
+from ..graph_db import get_current_transaction
+from ..models.node_models import EntryNode, EntryNodeSearchResult
+
+
+def get_query_param_name(index: int) -> str:
+    return f"value_{index}"
+
+
+class FilterSearchTerm(ABC):
+    @abstractmethod
+    def build_cypher_query(self, index: int) -> tuple[str, str | None]:
```

Comment:
please add a docstring to explain how it works ! And what it returns (why a tuple ?), and why we give the index.

Note: (easy to say after reading your code) we cloud instead return a dict as second part of the tuple, so that we may support multiple parameters or none easily. To build our parameter dict we wust have to collect those dict into one. (index parameter is still important to ensure distinct parameters).